src/flp/types.rs

```diff
@@ -1484,6 +1484,115 @@ mod tests {
+    #[derive(Debug, Clone, PartialEq, Eq)]
+    struct HigherDegree;
+
+    impl HigherDegree {
+        fn new() -> Self {
+            Self
+        }
+    }
+
+    impl Flp for HigherDegree {
+        type Field = TestField;
+
+        fn gadget(&self) -> Vec<Box<dyn Gadget<Self::Field>>> {
+            vec![Box::new(PolyEval::new(
+                vec![
+                    TestField::from(0),
+                    TestField::from(2),
+                    -TestField::from(3),
+                    TestField::from(1),
+                ],
+                1,
+            ))]
+        }
+
+        fn num_gadgets(&self) -> usize {
+            1
+        }
+
+        fn valid(
+            &self,
+            gadgets: &mut Vec<Box<dyn Gadget<Self::Field>>>,
+            input: &[Self::Field],
+            joint_rand: &[Self::Field],
+            _num_shares: usize,
+        ) -> Result<Vec<Self::Field>, FlpError> {
+            self.valid_call_check(input, joint_rand)?;
+            let check = gadgets[0].call(input)?;
+            Ok(vec![check])
+        }
+
+        fn input_len(&self) -> usize {
+            1
+        }
+
+        fn proof_len(&self) -> usize {
+            let gadget_arity = 1;
+            let gadget_degree = 3;
+            let gadget_calls = 1usize;
+            let p = (gadget_calls + 1).next_power_of_two();
+            gadget_arity + gadget_degree * (p - 1) + 1
+        }
+
+        fn verifier_len(&self) -> usize {
+            3
+        }
+
+        fn joint_rand_len(&self) -> usize {
+            0
+        }
+
+        fn eval_output_len(&self) -> usize {
+            1
+        }
+
+        fn prove_rand_len(&self) -> usize {
+            1
+        }
+    }
+
+    impl Type for HigherDegree {
+        type Measurement = u64;
+
+        type AggregateResult = u64;
+
+        fn encode_measurement(
+            &self,
+            measurement: &Self::Measurement,
+        ) -> Result<Vec<Self::Field>, FlpError> {
+            Ok(vec![TestField::from(*measurement)])
+        }
+
+        fn truncate(&self, input: Vec<Self::Field>) -> Result<Vec<Self::Field>, FlpError> {
+            self.truncate_call_check(&input)?;
+            Ok(input)
+        }
+
+        fn decode_result(
+            &self,
+            data: &[Self::Field],
+            _num_measurements: usize,
+        ) -> Result<Self::AggregateResult, FlpError> {
+            decode_result(data)
+        }
+
+        fn output_len(&self) -> usize {
+            1
+        }
+    }
+
+    #[test]
+    fn test_degree_3() {
+        let typ = HigherDegree::new();
+        TypeTest::expect_valid::<2>(&typ, &[TestField::from(0)], &[TestField::from(0)]);
+        TypeTest::expect_valid::<2>(&typ, &[TestField::from(1)], &[TestField::from(1)]);
+        TypeTest::expect_valid::<2>(&typ, &[TestField::from(2)], &[TestField::from(2)]);
+        TypeTest::expect_invalid::<2>(&typ, &[TestField::from(3)]);
```

Comment:
I suppose you could add a test with different share counts?
```suggestion
        TypeTest::expect_invalid::<2>(&typ, &[TestField::from(3)]);
        TypeTest::expect_valid::<3>(&typ, &[TestField::from(1)], &[TestField::from(1)]);
```